cylc/flow/scheduler.py

```diff
@@ -1562,6 +1556,30 @@ def start_job_submission(self, itasks: 'Iterable[TaskProxy]') -> bool:
         # one or more tasks were passed through the submission pipeline
         return True
 
+    def submit_task_jobs(
+        self, itasks: 'Iterable[TaskProxy]'
+    ) -> 'List[TaskProxy]':
+        """Prepare for job submission and submit task jobs.
+
+        Return: tasks that attempted submission.
+        """
+        # submit "simulation/skip" mode tasks, modify "dummy" task configs:
+        itasks, submitted_nonlive_tasks = (
+            self.task_job_mgr.submit_nonlive_task_jobs(
+                self.workflow, itasks, self.get_run_mode()
+            )
+        )
+
+        # submit "live" mode tasks (and "dummy" mode tasks)
+        submitted_live_tasks = self.task_job_mgr.submit_livelike_task_jobs(
+            self.workflow,
+            itasks,
+            self.server.curve_auth,
+            self.server.client_pub_key_dir,
+        )
```

Comment:
I'm not a big fan of moving this task submission logic into the scheduler.

I'm guessing you did this to avoid having to populate the extra args in integration tests, fair enough.

Maybe just leave a shim in scheduler.py to do this, would rather keep submission logic in one place, Cylc logic has a habit of sprawling.

Reply:
https://github.com/cylc/cylc-flow/compare/378e65b97cd5fefb123a523b204d2057057c50e9..d71cd087a6d1a7e7e93dbbf9c3a0fbdf83c9bbb7